### PR TITLE
Fix recreating Certificate bug

### PR DESCRIPTION
**Pull request by binoue**

### CODE REVIEW

File: test/cert-manager_test.go [hunk 1]
```diff
@@ -167,15 +167,24 @@ OUTER:
...
-	reason := targetCertReq.Status.Conditions[0].Reason
-	if reason == "Failed" {
-		log.Error("CertificateRequest failed", map[string]interface{}{
-			"certificate name":         cert.Name,
-			"certificate request name": targetCertReq.Name,
-		})
-		return true, nil
+	if len(targetCertReq.Status.Conditions) == 0 {
+		return false, errors.New("status not found")
 	}
```
Comment (ymmt2005) [inline test/cert-manager_test.go]: don't need.

File: test/cert-manager_test.go [hunk 2]
```diff
@@ -167,15 +167,24 @@ OUTER:
...
-	return false, fmt.Errorf("CertificateRequest is not ready")
+
+	for _, st := range targetCertReq.Status.Conditions {
+		if st.Type != certmanagerv1alpha2.CertificateRequestConditionReady {
+			continue
+		}
+		reason := st.Reason
+		if reason == "Failed" {
```
Comment (ymmt2005) [inline test/cert-manager_test.go]: Why don't you reference `Status`?
https://godoc.org/github.com/jetstack/cert-manager/pkg/apis/meta/v1#ConditionStatus

Comment (binoue) [reply test/cert-manager_test.go]: As shown in [here](https://cert-manager.io/docs/concepts/certificaterequest/#conditions), status can be False when the reason is InProgress. Thinking of that, I thought it is better to  recreate Certificate only when its reason is Failed.

# The above document column "Ready" is [this](https://github.com/jetstack/cert-manager/blob/abb680756e84b56df47f85a8ce732a0e09e0f171/deploy/manifests/00-crds.yaml#L3-L12)

The possible reason values are as follows
```
const (
	CertificateRequestReasonPending = "Pending"
	CertificateRequestReasonFailed  = "Failed"
	CertificateRequestReasonIssued  = "Issued"
)
```
https://github.com/jetstack/cert-manager/blob/master/pkg/apis/certmanager/v1alpha2/types_certificaterequest.go#L25-L29

Comment (ymmt2005) [reply test/cert-manager_test.go]: So, why don't you use that constant?

File: test/cert-manager_test.go (hunk 1 above)
Comment (ymmt2005) [reply test/cert-manager_test.go]: not fixed yet.

Comment (binoue) [reply test/cert-manager_test.go]: Sorry.. I did not push my local commit..

File: test/cert-manager_test.go [hunk 3]
```diff
@@ -167,15 +163,19 @@ OUTER:
...
-		log.Error("CertificateRequest failed", map[string]interface{}{
-			"certificate name":         cert.Name,
-			"certificate request name": targetCertReq.Name,
-		})
-		return true, nil
+	for _, st := range targetCertReq.Status.Conditions {
+		if st.Type != certmanagerv1alpha2.CertificateRequestConditionReady {
+			continue
+		}
+		if st.Status == certmanagerv1alpha2.CertificateRequestReasonFailed {
```
Comment (ymmt2005) [inline test/cert-manager_test.go]: `st.Reason`

**Pull request merged**